Fixes #8 Added Node Slack Bot

shivenmian opened the pull request:
@sudheesh001 @mariobehling Added slack bot. As of now it runs only when I start it up from my terminal. Will deploy to heroku later.

sudheesh001 commented:
> Hardcoding the trigger word as "Susi" made me a bit uncomfortable. 

Why is the hardcoding required ? Is this an issue with the API ? I don't find the same problem when I use the susi API and see the responses

shivenmian commented:
@sudheesh001 The thing is, the API has to be called only when the message starts with "Susi". So when I set up the webhook, there was an option for "Trigger Words", i.e it should POST to the API when the message starts with the Trigger word, so I added "Susi" as trigger word.

Now that I think of it, I can just evaluate ```req.body.text``` and split it and check if the message starts with Susi or not. But it's still working with keeping the Trigger word setting.

Is this fine or do I add the ```req.body.text contains Susi``` check in the code?

sudheesh001 commented:
Why can't we use `mentions` like `@susi` for the same, can we use every mentioned message of `susi` to trigger rather than reading every message and discarding the ones we don't need ?

mariobehling commented:
Yes, `@susi!` seems the more obvious choice.

sudheesh001 commented on slack/index.js:
```diff
@@ -0,0 +1,42 @@
+//startup
+app.listen(app.get('port'), function() {
+	console.log('running on port', app.get('port'));
+});
```
Break new line here please.

shivenmian commented:
Okay, fixing this.

shivenmian commented:
@sudheesh001 @mariobehling To make it better, I used a Slack API wrapper to write this. I'm also not using a webhook anymore. All I do is run the code from my terminal and the bot runs.

The only drawback is that in line 27 of index.js, I had to hardcode my bot id. This was because the wrapper I was using had an error in the method they said would return the bot id (i'd checked their code), so I could not use it.

The bot runs from my terminal as of now. Will deploy onto Heroku later. Please check.

Screenshot:
<img width="1046" alt="screen shot 2016-08-05 at 2 02 14 am" src="https://cloud.githubusercontent.com/assets/14081800/17417373/abdfa572-5ab0-11e6-8f3a-67f49ca1d645.png">

sudheesh001 commented:
Needs refactoring as discussed and removal of merge conflicts.

shivenmian commented:
Yes, will fix.
